src/core/server/saved_objects/service/lib/utils.ts

```diff
@@ -69,4 +71,21 @@ export class SavedObjectsUtils {
     total: 0,
     saved_objects: [],
   });
+
+  /**
+   * Generates a random ID for a saved objects.
+   */
+  public static generateId() {
+    return uuid.v1();
+  }
+
+  /**
+   * Validates that a saved object ID matches UUID format.
+   *
+   * @param {string} id The ID of a saved object.
+   * @todo Use `uuid.validate` once upgraded to v5.3+
+   */
+  public static isRandomId(id: string | undefined) {
+    return typeof id === 'string' && UUID_REGEX.test(id);
+  }
```

Comment:
I think this naming is misleading. We are asserting that the id matches the uuid format, not that is was randomly generated (which is properly explained in the tsdoc, but not really reflected on the method name)

Reply:
The purpose of the `isRandomId` method is to allow plugin developers to assert whether an ID has been randomly generated or whether it's a static ID. In order to do so the function checks whether a string matches UUID format but that's an implementation detail that consumers don't need to be aware of in the same way that they don't need to know that `generateId` generates a UUID. 

Would `isRandomlyGeneratedId` be more explicit?

Reply:
That would just be a rephrasing.

In that case, change `Validates that a saved object ID matches UUID format.` to `Validates that a saved object ID has been randomly generated`